R/transformCounts.R

```diff
@@ -779,11 +793,96 @@ setMethod("transformAssay", signature = c(x = "SingleCellExperiment"),
     return(mat)
 }
 
+########################### .apply_transformation_difference ##################
+# Computes all pairwise differences (x - y) between features across samples.
+# Returns a matrix with one row per feature pair.
+.apply_transformation_difference <- function(mat, ...) {
+    # Check that rownames are present
+    if( is.null(rownames(mat)) ) {
+        warning("No rownames found in the matrix — generated names will be used.")
+        rownames(mat) <- paste0("Feature", seq_len(nrow(mat)))
+    }
+  
+    # Get the feature (taxa) names
+    taxa <- rownames(mat)
+  
+    # Generate all unique pairwise combinations
+    combs <- utils::combn(taxa, 2, simplify = FALSE)
```

Comment:
Add

#' @importFrom stats combn

and use comb() without defining the package